examples/MQ/serialization/sampler.cxx

```diff
@@ -64,6 +66,9 @@ struct Sampler : fair::mq::Device
             ExHeader header;
             header.eventNumber = idx;
 
+            // temporarily apply data processing steps to check that they work before & after serialization
+            TClonesArray hits = FindHits(*fInput);
```

Comment:
I hope the optimizer is not removing this completely… Is there some construct in C++ to let the compiler think that this local variable is used?

Reply:
Not sure. I could add a log message with the number of hits found.

Reply:
I remember https://github.com/google/benchmark/blob/main/docs/user_guide.md#preventing-optimization, maybe the implementation of this helper reveals something useful.

Reply:
How about `volatile`?

Reply:
Looks like the benchmark uses `volatile`: https://github.com/google/benchmark/blob/main/include/benchmark/benchmark.h#L444
But what exactly the rest does I don't understand yet.

Reply:
I think, the magic is in the `asm`. AFAIK the compiler does not look at the assembler code at all. It just knows that the developer wants to do *something*. (We know, it's nothing, because we can look at the `""`.)

Guessing: the `"+m,r"(value)` might tell the compiler, that the assembler code needs `value` as input for read and/or write access and should give the assembler code access to it in some register or something. That way the compiler has to provide `value` (because it doesn't know what the assembler code actually does).